static/app/views/alerts/list/rules/teamFilter.tsx

```diff
@@ -98,7 +98,13 @@ function TeamFilter({
       }
       value={selectedTeams}
       onInputChange={debounce(val => void onSearch(val), DEFAULT_DEBOUNCE_DURATION)}
-      onChange={opts => handleChangeFilter(opts.map(opt => opt.value))}
+      onChange={opts => {
+        // Compact select type inference does not work - onChange type is actually T | null.
+        if (!opts) {
+          return handleChangeFilter([]);
+        }
+        return handleChangeFilter((opts ?? []).map(opt => opt.value));
```

Comment:
Feels like opts should be defined with the new check above

Reply:
What do you mean by that?

Reply:
Oh you mean in the onChange? Maybe, but I wanted to keep it closer to CompactSelect because it should really be fixed there

Reply:
just the `opts ?? []` its already checking for `!opts` so opts has to be defined

Reply:
Oh right, yeah that makes no sense

Reply:
I think I first did that and then decided to use an `if` and add the comment. Good catch!